### DISCUSSION TITLE
[Magiclink] Support for Fediverse and Matrix mentions

### DISCUSSION THREAD

**Discussion by Andre601** [facelessuser/pymdown-extensions#2381] [Ideas]
Unsure if this is worth the effort of implementing into Magiclink (I'm tempted to make an extension myself but lack the experience. And looking at Magiclink for inspiration doesn't help me too much either 😅)

The suggestion is to add support for the mention patterns of fediverse- and matrix-based accounts using the `@username@domain.tld` and `@username:domain.tld` respectively, so that f.e. `@mastodon@mastodon.social` is parsed into `<a href="https://mastodon.social/@mastodon">@mastodon@mastodon.social</a>`

**Comment by facelessuser** [comment]
While I'm certainly open to adding additional, very popular social media support, Mastadon, and those like it, do pose a bit of a problem as there are multiple, distributed servers and not one site. With that said, I could see a potential of allowing people to define specific shortnames for a given Mastadon domain: `@mastodon_social:user` -> `<a href="https://mastodon.social/@mastodon">@mastodon</a>`.

**Reply by Andre601** [reply]
> Mastadon, and those like it, do pose a bit of a problem as there are multiple, distributed servers and not one site.

Where is the issue exactly? The pattern is `@<user>@<domain>`, which I also propose to be used here, which already includes the instance alongside the username.
Only real downside is, that some sites may not have the `domain.tld/@username` pattern used, but from what I know is that very rare as it's a standard to have it that way...

**Reply by facelessuser** [reply]
> Where is the issue exactly? The pattern is @<user>@<domain>, which I also propose to be used here, which already includes the instance alongside the username.

I'm not sure I like this syntax and it complicates the internal handling. The generic format is `@user` (for whatever your default assumed site is) or `@site:user` for explicitly differentiating sites. User name is what we will have appear. We also don't allow any and all characters in the site name, so a friendly alias for easy parsing is desired, no `.` in site names.

Ideally, a user could define a custom social name with a host alias of their choice, and then use the conventional, generic syntax we already provide.

**Reply by facelessuser** [reply]
Essentially, it would follow the pattern we have set with custom repository hosts: https://facelessuser.github.io/pymdown-extensions/extensions/magiclink/#custom-repository-hosts
